tasks/copy_to_glacier/sqs_library.py

```diff
@@ -0,0 +1,153 @@
+        def wrapper_retry_error(*args, **kwargs) -> RT:
+            """
+            Wrapper that performs our extra tasks on the function
+            """
+            # Initialize the retry loop
+            total_retries = 0
+
+            # Enter loop
+            while True:
+                # Try the function and catch the expected error
+                try:
+                    return func(*args, **kwargs)
+                except Exception:
+                    if total_retries == max_retries:
+                        # Log it and re-raise if we maxed our retries + initial attempt
+                        LOGGER.error(
+                            "Encountered Error {total_attempts} times. Reached max retry limit.",
+                            total_attempts=total_retries,
+                        )
+                        raise
+                    else:
+                        # perform exponential delay
+                        backoff_time = (
+                            backoff_in_seconds * backoff_factor ** total_retries
+                            + random.uniform(0, 1)  # nosec
+                        )
+                        LOGGER.error(
+                            "Encountered Error on attempt {total_attempts}. "
+                            "Sleeping {backoff_time} seconds.",
+                            total_attempts=total_retries,
+                            backoff_time=backoff_time,
+                        )
+                        time.sleep(backoff_time)
+                        total_retries += 1
+
+        # Return our wrapper
+        return wrapper_retry_error
+
+    # Return our decorator
+    return decorator_retry_error
+
+
+def get_aws_region() -> str:
+    """
+    Gets AWS region variable from the runtime environment variable.
+        Returns:
+            The AWS region variable.
+        Raises:
+            Exception: Thrown if AWS region is empty or None.
+    """
+    LOGGER.debug("Getting environment variable AWS_REGION value.")
+    aws_region = os.getenv("AWS_REGION", None)
+    if aws_region is None or len(aws_region) == 0:
+        message = "Runtime environment variable AWS_REGION is not set."
+        LOGGER.critical(message)
+        raise ValueError(message)
+    LOGGER.debug(f"Got environment variable for AWS_REGION = {aws_region}")
+    return aws_region
+
+
+@retry_error()
+def post_to_metadata_queue(
+    sqs_body: Dict[str, Any],
+    metadata_queue_url: str,
+) -> None:
+    """
+    Posts metadata information to the metadata SQS queue.
+    Args:
+        sqs_body: A dictionary containing the metadata objects that will be sent to SQS.
+        db_queue_url: The metadata SQS queue URL defined by AWS.
+    Raises:
+        None
+    """
+    # validate body.json schema
+    with open("schemas/body.json", "r") as raw_schema:
+        schema = json.loads(raw_schema.read())
+    validate = fastjsonschema.compile(schema)
```

Comment:
I would put a debug logging statement in here to note to the end user that you are performing a validation of the message to help with troubleshooting.